controller/lib/hal/hal_stm32.cpp

```diff
@@ -292,6 +287,57 @@ void HalApi::delay(Duration d) {
 
 Time HalApi::now() { return millisSinceStartup(msCount); }
 
+/******************************************************************
+ * Loop timer
+ *
+ * I use one of the basic timers (timer 7) to generate the interrupt
+ * from which the high priority loop function is called.
+ *
+ * The basic timers (like timer 7) are documented in chapter 29 of
+ * the reference manual
+ *****************************************************************/
+static void (*high_priority_callback)();
+void HalApi::startLoopTimer(const Duration &period, void (*callback)()) {
+  high_priority_callback = callback;
+
+  // Find the loop period in clock cycles
+  int32_t reload = static_cast<int32_t>(CPU_FREQ * period.seconds());
+  int prescale = 1;
+
+  // Adjust the prescaler so that my reload count will fit in the 16-bit
+  // timer.
+  if (reload > 65536) {
+    prescale = static_cast<int>(reload / 65536.0) + 1;
+    reload /= prescale;
+  }
+
+  // Enable the clock to the timer
+  EnableClock(TIMER15_BASE);
+
+  // Just set the timer up to count every microsecond.
+  TimerRegs *tmr = TIMER15_BASE;
+  tmr->reload = reload - 1;
+  tmr->prescale = prescale - 1;
+  tmr->event = 1;
+  tmr->ctrl[0] = 1;
+  tmr->intEna = 1;
+
+  // Enable the interrupt with a higher priority value then
```

Comment:
I think it would be worthwhile to add a well-documented enum containing the priorities of different interrupts we use, so that there's one place in code where one can understand which ones are higher priority than others - as opposed to having to search the code for other EnableInterrupt() calls and seeing whose value is higher or lower than 8.

Reply:
Yes, good idea.  Done.